src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForSchemaReadTests.cs

```diff
@@ -0,0 +1,410 @@
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+Imports System.IO
+Imports System.Xml.Schema
+Class TestClass
+    Private Sub TestMethod()
+        Try
+            Dim src = """"
+            Dim tr As TextReader = New StreamReader(src)
+            Dim schema As XmlSchema = XmlSchema.Read(tr, Nothing)
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075BasicResultAt(10, 39, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadInCatchBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+   using System;
+   using System.Data;
+using System.IO;
+using System.Xml.Schema;
+    class TestClass
+    {
+        private void TestMethod()
+        {
+            try { }
+            catch (Exception)
+            {
+                var src = """";
+                TextReader tr = new StreamReader(src);
+                XmlSchema schema = XmlSchema.Read(tr, null);
+            }
+            finally { }
+        }
+    }",
+                GetCA3075CSharpResultAt(15, 36, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+Imports System.IO
+Imports System.Xml.Schema
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Dim src = """"
+            Dim tr As TextReader = New StreamReader(src)
+            Dim schema As XmlSchema = XmlSchema.Read(tr, Nothing)
+        Finally
+        End Try
+    End Sub
+End Class",
+                GetCA3075BasicResultAt(11, 39, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadInFinallyBlockShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+   using System;
+   using System.Data;
+using System.IO;
+using System.Xml.Schema;
+    class TestClass
+    {
+        private void TestMethod()
+        {
+            try { }
+            catch (Exception) { throw; }
+            finally
+            {
+                var src = """";
+                TextReader tr = new StreamReader(src);
+                XmlSchema schema = XmlSchema.Read(tr, null);
+            }
+        }
+    }",
+                GetCA3075CSharpResultAt(16, 36, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Data
+Imports System.IO
+Imports System.Xml.Schema
+Class TestClass
+    Private Sub TestMethod()
+        Try
+        Catch generatedExceptionName As Exception
+            Throw
+        Finally
+            Dim src = """"
+            Dim tr As TextReader = New StreamReader(src)
+            Dim schema As XmlSchema = XmlSchema.Read(tr, Nothing)
+        End Try
+    End Sub
+End Class",
+                GetCA3075BasicResultAt(13, 39, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadInAsyncAwaitShouldGenerateDiagnostic()
```

Comment:
It's not the `async` that's relevant, it's the fact that the warning occurs in a lambda, whose nearest enclosing named method (in this case) is `Run`. If you were to keep this test (which you won't, assuming you remove the method names from the warning messages), this test might be better named `UseXmlSchemaReadInLambdaPassedToNamedFunction`.